### PR TITLE
Add the templ parser

### PR DESCRIPTION
**Pull request by vrischmann**
This adds the [tree-sitter-templ](https://github.com/vrischmann/tree-sitter-templ) parser.

### CODE REVIEW

**Comment by amaanq** [comment]
Are you unable to inherit go's queries?

**Comment by vrischmann** [comment]
> Are you unable to inherit go's queries?

Do you mean with this
```
; inherits: lang1,(optionallang)
```
as described in [CONTRIBUTING.md](https://github.com/nvim-treesitter/nvim-treesitter/blob/master/CONTRIBUTING.md) ?
I haven't tried it yet, I'll have to check.

**Comment by vrischmann** [comment]
(I'm documenting this here so others may find it; it's hard to find any relevant info searching for this error).

Ok so I tried to use `; inherits: go` and at first I was only getting an error like this:
```
ERR 2023-11-13T22:44:59.606 nvim.41136.0 decor_provider_invoke:47: error in provider treesitter/highlighter:line: Error executing lua: /usr/share/nvim/runtime/lua/vim/treesitter/query.lua:260: query: invalid node type at position 1266 for language templ
stack traceback:
	[C]: in function '_ts_parse_query'
	/usr/share/nvim/runtime/lua/vim/treesitter/query.lua:260: in function 'get'
	/usr/share/nvim/runtime/lua/vim/treesitter/highlighter.lua:50: in function 'new'
	/usr/share/nvim/runtime/lua/vim/treesitter/highlighter.lua:202: in function 'get_query'
	/usr/share/nvim/runtime/lua/vim/treesitter/highlighter.lua:228: in function 'fn'
	/usr/share/nvim/runtime/lua/vim/treesitter/languagetree.lua:345: in function 'for_each_tree'
	/usr/share/nvim/runtime/lua/vim/treesitter/highlighter.lua:214: in function 'on_line_impl'
	/usr/share/nvim/runtime/lua/vim/treesitter/highlighter.lua:290: in function </usr/share/nvim/runtime/lua/vim/treesitter/highlighter.lua:284>
```

I struggled to understand the problem but after some digging I realized that the queries in `nvim-treesitter` were not compatible with the version of `tree-sitter-go` I was using in my parser (0.19.1).

I updated `tree-sitter-go` to 0.20.0, fixed a bunch of things and now it works.

The queries files now only contain things that are specific to templ.

File: queries/templ/injections.scm [hunk 1]
```diff
@@ -0,0 +1,6 @@
+; inherits: go
+
```
Comment (ObserverOfTime) [inline queries/templ/injections.scm:2]:
```suggestion

((element_comment) @injection.content (#set! injection.language "comment"))

```

File: queries/templ/highlights.scm [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+(component_import
+  name: (component_identifier) @function)
+
+(component_render) @function.call
+
+(element_comment) @comment @spell
+
+[
+  "@"
+] @operator
```
Comment (ObserverOfTime) [inline queries/templ/highlights.scm:37]:
```suggestion
"@" @operator
```

File: queries/templ/highlights.scm [hunk 3]
```diff
@@ -0,0 +1,43 @@
+; inherits: go
+
+(component_declaration
+  name: (component_identifier) @function)
+
+(tag_start) @tag
+(tag_end) @tag
+(self_closing_tag) @tag
+(style_element) @tag
```
Comment (ObserverOfTime) [inline queries/templ/highlights.scm:9]:
```suggestion
[
  (tag_start)
  (tag_end)
  (self_closing_tag)
  (style_element)
] @tag
```

File: queries/templ/highlights.scm [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+
+(attribute
+  name: (attribute_name) @tag.attribute)
+(attribute
+  value: (quoted_attribute_value) @string)
+
+(element_text) @string.special
+(style_element_text) @string.special
```
Comment (ObserverOfTime) [inline queries/templ/highlights.scm:17]:
```suggestion
[
  (element_text)
  (style_element_text)
] @string.special
```

File: queries/templ/highlights.scm [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+
+(css_identifier) @function
+(css_property
+  name: (css_property_name) @attribute)
+(css_property
+  value (css_property_value) @attribute)
+
+(expression) @function.method
+(dynamic_class_attribute_value) @function.method
```
Comment (ObserverOfTime) [inline queries/templ/highlights.scm:26]:
```suggestion
[
  (expression)
  (dynamic_class_attribute_value)
] @method
```

**Review by ObserverOfTime — CHANGES_REQUESTED** [review]

**Review by ObserverOfTime — APPROVED** [review]

**Comment by vrischmann** [comment]
There was an error in one of the query which made the checks fail, it's fixed now.

**Review by amaanq — APPROVED** [review]

**Pull request merged**